### PR TITLE
Traffic Ops client upgrades

### PR DESCRIPTION
**Pull request by ocket8888**
## What does this PR (Pull Request) do?
Updates various clients of the Traffic Ops API to use version 2.0.

- [x] This PR is not related to any Issue

## Which Traffic Control components are affected by this PR?
This includes updates to all of the following components/systems/tools:

- experimental Traffic Portal in Angular 8
- CDN in a Box, including ORT.py (except for configuration file generation routes)
- The `astats` plugin simulator in `test/astats_sim/`
- The `test/traffic_ops_cfg/cfg_test.pl` script - which by the way can we remove that? Isn't it superseded by the `compare` tool?
- The `compare` tool and its companion `genConfigRoutes.py` script - the latter of which should be removed with API v1 anyway, since it will no longer be able to generate valid routes (because configuration file generation will be handled by `atstccfg` from now on).
- Traffic Ops Client (Java)
- Some of the 'checks' scripts in `traffic_ops/app/bin/checks/`
- The `traffic_ops/app/bin/extensions` script
- The `traffic_ops/app/script/update_riak_for_search.pl` script

This PR also removes API endpoints from clients when said endpoints don't exist in TO API v2.0.

It also fixes a typo in one of the TO route definitions and some improper formatting for the `--api-routes` option of `traffic_ops_golang`.

## What is the best way to verify this PR?
For tools/components/systems that support it, run the tests. For those that don't....  ¯\\\_(ツ)\_/¯

## The following criteria are ALL met by this PR
- [x] Tests will be added in #4374 
- [x] Documentation will be added in #4371 
- [x] This PR includes an update to CHANGELOG.md
- [x] This PR includes any and all required license headers
- [x] This PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY**

### CODE REVIEW

**Comment by ocket8888** [comment]
This was supposed to be a draft...

File: CHANGELOG.md [hunk 1]
```diff
@@ -26,8 +26,12 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 ### Changed
 - Fix to traffic_ops_ort.pl to strip specific comment lines before checking if a file has changed.  Also promoted a changed file message from DEBUG to ERROR for report mode.
+- Updated The Traffic Ops Python, Go and Java clients to use API version 2.0 (when possible)
+- Updated CDN-in-a-Box scripts and enroller to use TO API version 2.0
+- Updated numerous, miscellaneous tools to use TO API version 2.0
 
 ### Deprecated/Removed
+- Traffic Ops Python client no longer supports Python 2.
```
[the remark this answers is not in the document]
Comment (mhoppa) [reply CHANGELOG.md]: or I guess a better question is can this be in another PR?

[the remark this answers is not in the document]
Comment (rawlinp) [reply CHANGELOG.md]: That is at least one thing that should be split off into multiple separate PRs IMO -- one PR to make all the routes use 2.0, removing methods for deprecated 1.x routes and a second PR to remove the python2-isms.

9120 LOC change is too big for a single PR.

[the remark this answers is not in the document]
Comment (ocket8888) [reply CHANGELOG.md]: Well I've got the Python client upgrades in a separate PR now, probably gonna do the same for TP, and then another one for "misc".

**Pull request merged**